File: solutions/70.climbing-stairs.py
#
# @lc app=leetcode id=70 lang=python3
#
# [70] Climbing Stairs
#

# @lc code=start
class Solution:
    def climbStairs(self, n: int) -> int:
        dp = [1] + [0] * n

        # n = n if n <= 2
        # n = n(n-1) + n(n-2) if n > 2
        for i in range(1, n + 1):
            dp[i] += dp[i - 1] if i >= 1 else 0
            dp[i] += dp[i - 2] if i >= 2 else 0

        return dp[n]
# @lc code=end

File: solutions/70.climbing-stairs.ts
/*
 * @lc app=leetcode id=70 lang=typescript
 *
 * [70] Climbing Stairs
 */

// @lc code=start
function climbStairs(n: number): number {
  const dp: number[] = [1, ...[...Array(n)].map(_ => 0)]

  for(let i = 0; i < n + 1; i++){
    dp[i] += (i >= 1 ? dp[i - 1] : 0)
    dp[i] += (i >= 2 ? dp[i - 2] : 0)
  }

  return dp[n]
};
// @lc code=end
